### PR TITLE
move functions from src/functions into static methods of entities

### PR DESCRIPTION
**Pull request by nguyenvukhang**

### CODE REVIEW

File: packages/database/src/entity/.eslintrc.js [hunk 1]
```diff
@@ -9,6 +9,17 @@ module.exports = {
...
     ...config.rules,
     'new-cap': 'off',
-    'require-jsdoc': 'off',
+    'require-jsdoc': [
+      'error',
+      {
+        require: {
+          FunctionDeclaration: true,
+          MethodDefinition: true,
+          ClassDeclaration: false,
```
Comment (nguyenvukhang) [inline packages/database/src/entity/.eslintrc.js:18]: `ClassDeclaration: false` is the only difference from the default settings

File: packages/database/src/entity/Module.ts [hunk 2]
```diff
@@ -104,4 +97,104 @@ export class Module {
     m.fulfillRequirements = props.fulfillRequirements || []
     return m
   }
+
```
Comment (nguyenvukhang) [inline packages/database/src/entity/Module.ts:100]: This chunk is what used to be in `src/functions/module.ts`

File: packages/database/types/modtree.d.ts [hunk 3]
```diff
@@ -25,3 +25,12 @@ export type DegreeProps = {
   modulesRequired: Module[]
   title: string
 }
+
```
Comment (nguyenvukhang) [inline packages/database/types/modtree.d.ts:28]: a nice overview of all classes and their functions

**Review by weiseng18 — APPROVED** [review]
nice refactor, thanks for the comments in the PR

**Pull request merged**